lexicon/providers/dreamhost.py

```diff
@@ -0,0 +1,181 @@
+            raise Exception('unable to get records: %s' % payload)
+
+        resource_list = [
+            resource for resource in resource_list if resource['zone'] == self.domain]
+        if rtype:
+            resource_list = [
+                resource for resource in resource_list if resource['type'] == rtype]
+        if name:
+            name = self._full_name(name)
+            resource_list = [
+                resource for resource in resource_list if resource['record'] == name]
+        if content:
+            resource_list = [
+                resource for resource in resource_list if resource['value'] == content]
+
+        processed_records = []
+        for resource in resource_list:
+            processed_records.append({
+                'id': resource['type'] + '|' + resource['record'] + '|' + resource['value'],
+                'type': resource['type'],
+                'name': resource['record'],
+                'content': resource['value'],
+            })
+
+        return processed_records
+
+    # Create or update a record.
+    def _update_record(self, identifier, rtype=None, name=None, content=None):
+        if identifier == '':
+            identifier = None
+        if identifier is not None:
+            id_list = identifier.split('|')
+            if len(id_list) != 3:
+                raise Exception('invalid identifier: %s' % (identifier))
+
+            id_type, id_name, id_content = id_list
+
+            try:
+                self._delete_record(rtype=id_type, name=id_name, content=id_content)
+            except:  # pylint: disable=bare-except
+                pass
+
+        return self._create_record(rtype=rtype, name=name, content=content)
+
+    # Delete an existing record.
+    # If record does not exist, do nothing.
+    def _delete_record(self, identifier=None, rtype=None, name=None, content=None):
+
+        id_type = None
+        id_name = None
+        id_content = None
+        if identifier is not None:
+            id_list = identifier.split('|')
+            if len(id_list) != 3:
+                raise Exception('invalid identifier: %s' % (identifier))
+
+            id_type, id_name, id_content = id_list
+
+        if rtype is None:
+            rtype = id_type
+        if name is None:
+            name = id_name
+        if content is None:
+            content = id_content
+
+        if name is not None:
+            name = self._full_name(name)
+
+        to_deletes = []
+        if rtype is None or name is None or content is None:
+            if identifier is None:
+                records = self._list_records(rtype=rtype, name=name, content=content)
+                to_deletes = records
+        else:
+            to_deletes = [{'record': name, "type": rtype, "value": content}]
+
+        for each in to_deletes:
+            try:
+                self._get('dns-remove_record', query_params=each)
+            except:  # pylint: disable=bare-except
+                pass
```

Comment:
Can you explain what is the expected exception here, and why it is not a problem ?

Reply:
I am not sure about the policy that what if we successfully delete 1 record but fail 1 another record and there are 2 remaining to-delete-records in the for-loop.
(How to deal with the 2 remaining to-delete-records?)

I think I'll change to the following setup:
1. for-loop all the candidate-to-delete-record
2.      catch the exception as err.
3. after for-loop, if err is not None: raise err (catch the last exception)

Reply:
Yes it is a good approach.